src/devices/Vl53L1X/Vl53L1X.cs

```diff
@@ -0,0 +1,1022 @@
+
+        /// <summary>
+        ///     This function returns the current number of enabled SPADs.
+        /// </summary>
+        /// <returns>the current number of enabled SPADs.</returns>
+        public ushort GetSpadNb()
+        {
+            return (ushort)(ReadUInt16((ushort)Registers.VL53L1X_RESULT__DSS_ACTUAL_EFFECTIVE_SPADS_SD0) >> 8);
+        }
+
+        /// <summary>
+        ///     This function returns the ambient rate in kcps.
+        /// </summary>
+        /// <returns>The ambient rate in kcps.</returns>
+        public ushort GetAmbientRate()
+        {
+            return (ushort)(ReadUInt16((ushort)Registers.RESULT__AMBIENT_COUNT_RATE_MCPS_SD) * 8);
+        }
+
+        /// <summary>
+        ///     This function returns the <see cref="RangeStatus" /> of the device.
+        /// </summary>
+        /// <returns>The <see cref="RangeStatus" /></returns>
+        public RangeStatus GetRangeStatus()
+        {
+            byte rgSt = ReadByte((ushort)Registers.VL53L1X_RESULT__RANGE_STATUS);
+
+            rgSt = (byte)(rgSt & 0x1F);
+
+            switch (rgSt)
+            {
+                case 9:
+                    return RangeStatus.NoError;
+                case 6:
+                    return RangeStatus.SigmaFailure;
+                case 4:
+                    return RangeStatus.SignalFailure;
+                case 5:
+                    return RangeStatus.OutOfBounds;
+                case 7:
+                    return RangeStatus.WrapAround;
+                default:
+                    throw new Exception($"The returned range status code {rgSt} of the device is unknown.");
+            }
+        }
+
+        /// <summary>
+        ///     This function programs the offset correction in mm.
+        /// </summary>
+        /// <param name="offsetValue">The offset correction value to program in mm</param>
+        public void SetOffset(short offsetValue)
+        {
+            WriteInt16((ushort)Registers.ALGO__PART_TO_PART_RANGE_OFFSET_MM, (short)(offsetValue * 4));
+            WriteInt16((ushort)Registers.MM_CONFIG__INNER_OFFSET_MM, 0x0);
+            WriteInt16((ushort)Registers.MM_CONFIG__OUTER_OFFSET_MM, 0x0);
+        }
+
+        /// <summary>
+        ///     This function returns the programmed offset correction value in mm.
+        /// </summary>
+        /// <returns>The offset correction value in mm.</returns>
+        public short GetOffset()
+        {
+            short offset = ReadInt16((ushort)Registers.ALGO__PART_TO_PART_RANGE_OFFSET_MM);
+            offset <<= 3;
+            offset /= 32;
+            return offset;
+        }
+
+        /// <summary>
+        ///     This function programs the crosstalk correction value in cps.
+        ///     This is the number of photons reflected back from the cover glass in cps.
+        /// </summary>
+        /// <param name="xTalkValue">The crosstalk correction value in cps.</param>
+        public void SetXtalk(ushort xTalkValue)
```

Comment:
and this one for a property as well

Reply:
I'm now using properties wherever possible